core/txpool/blobpool/blobpool.go

```diff
@@ -455,22 +444,26 @@ func (p *BlobPool) Close() error {
 // parseTransaction is a callback method on pool creation that gets called for
 // each transaction on disk to create the in-memory metadata index.
 func (p *BlobPool) parseTransaction(id uint64, size uint32, blob []byte) error {
-	item := new(blobTx)
-	if err := rlp.DecodeBytes(blob, item); err != nil {
+	tx := new(types.Transaction)
+	if err := rlp.DecodeBytes(blob, tx); err != nil {
 		// This path is impossible unless the disk data representation changes
 		// across restarts. For that ever unprobable case, recover gracefully
 		// by ignoring this data entry.
 		log.Error("Failed to decode blob pool entry", "id", id, "err", err)
 		return err
 	}
-	meta := newBlobTxMeta(id, size, item.Tx)
+	if tx.BlobSidecar() == nil {
+		return fmt.Errorf("missing blob sidecar in tx id %d", id)
```

Comment:
Please add a log.Error too, we want to detect this, it's a programming error.

Reply:
An I think teh return error should only be "missing blob sidecar", no need for the tx id, log will be visible anyway for test debugging